builtin/logical/database/path_roles.go

```diff
@@ -356,6 +371,23 @@ func (b *databaseBackend) pathRoleCreateUpdate(ctx context.Context, req *logical
 		if role.DBName == "" {
 			return logical.ErrorResponse("database name is required"), nil
 		}
+
+		if credentialTypeRaw, ok := data.GetOk("credential_type"); ok {
+			credentialType := credentialTypeRaw.(string)
+			if err := role.setCredentialType(credentialType); err != nil {
```

Comment:
Where do we set the default to be password if not specified?

Reply:
The zero value for `CredentialType` on the [roleEntry](https://github.com/hashicorp/vault/blob/db-secrets-credential-types/builtin/logical/database/path_roles.go#L614) is password. So if you don't provide the `credential_type`, it'll default to password.